onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc

```diff
@@ -33,7 +33,7 @@ bool BaseSelector::CheckQDQNodes(const Graph& graph, const Node& node,
 
   return num_dq_inputs == gsl::narrow_cast<int>(dq_nodes.size()) &&
          num_outputs == gsl::narrow_cast<int>(q_nodes.size()) &&
-         !graph.NodeProducesGraphOutput(node);
+         optimizer_utils::CheckOutputEdges(graph, node, 1);
```

Comment:
Why is removing the check that the node does not produce a graph output valid? If it produces a graph output we need that value with it's current type. This is not covered by the check on the number of edges.

Reply:
Sorry, I leave a comment yesterday but not sure why it doesn't appear here. Originally I remove graph.NodeProducesGraphOutput() because optimizer_utils::CheckOutputEdges() have the check. For the case you mentioned, could you give me an example that would generate `input(fp32) -> DQ -> Resize -> Q -> output`? IIUC, our tool would translate case like `input(fp32) -> Resize -> Reshape -> output` to `input(fp32) -> Q -> DQ -> Resize -> Q -> DQ -> Reshape -> output`. And it would transformed/optimized to `input(fp32) -> Q -> Resize -> Reshape -> DQ -> output` (like the test case I just added).

Reply:
Got it - thanks! I didn't know that CheckOutputEdges checked for graph outputs (function name could maybe be changed to make it clearer it's not just checking edges).

Call MakeOutput instead of MakeIntermediate for resize_output in your new test to create the Resize output and also make it a graph output.